evm_loader/program/src/executor.rs

```diff
@@ -309,28 +318,39 @@ impl<'config, B: Backend> Machine<'config, B> {
         Self{ executor, runtime }
     }
 
-    pub fn call_begin(&mut self, caller: H160, code_address: H160, input: Vec<u8>, gas_limit: u64) {
+    pub fn call_begin(&mut self,
```

Comment:
You should record transaction cost as well.
See how it is done in the original rust-evm - https://github.com/rust-blockchain/evm/blob/7a225c38dd5e308fe634e6606cd2a330dc1e24cc/src/executor/stack/mod.rs#L257

Reply:
A clarification is needed. We already have such code in our evm (branch 18-gasometer, of course): https://github.com/neonlabsorg/evm/blob/5b72bc771ef85750ad7fff056922a137592b38ea/src/executor/stack.rs#L212

Reply:
> You should record transaction cost as well.
> See how it is done in the original rust-evm - https://github.com/rust-blockchain/evm/blob/7a225c38dd5e308fe634e6606cd2a330dc1e24cc/src/executor/stack/mod.rs#L257

You should record transaction cost as well.
See how it is done in the original rust-evm - https://github.com/rust-blockchain/evm/blob/7a225c38dd5e308fe634e6606cd2a330dc1e24cc/src/executor/stack/mod.rs#L257

Reply:
Added recording of transaction cost in in create_begin & call_begin.